tests/tools/oecertdump/host/sgx_quote.cpp

```diff
@@ -111,6 +139,44 @@ void output_certificate_chain(const uint8_t* data, size_t data_len)
         result = ParseSGXExtensions(
             &leaf_cert, buffer, &buffer_size, &extension_info);
     }
+    printf(
+        "\n    parsed qe certificate extension (%s) {\n",
+        SGX_EXTENSION_OID_STR);
+    printf("        ppid (hex): ");
+    oe_hex_dump(extension_info.ppid, OE_COUNTOF(extension_info.ppid));
+    printf("        comp_svn (hex): ");
+    oe_hex_dump(extension_info.comp_svn, OE_COUNTOF(extension_info.comp_svn));
+    printf("        pce_svn: 0x%x\n", extension_info.pce_svn);
+    printf("        cpu_svn (hex): ");
+    oe_hex_dump(extension_info.cpu_svn, OE_COUNTOF(extension_info.cpu_svn));
+    printf("        pce_id (hex): ");
+    oe_hex_dump(extension_info.pce_id, OE_COUNTOF(extension_info.pce_id));
+    printf("        fmspc (hex): ");
+    oe_hex_dump(extension_info.fmspc, OE_COUNTOF(extension_info.fmspc));
+    printf("        sgx_type: %d\n", extension_info.sgx_type);
+    printf(
+        "        opt_dynamic_platform: %s\n",
+        extension_info.opt_dynamic_platform ? "true" : "false");
+    printf(
+        "        opt_cached_keys: %s\n",
+        extension_info.opt_cached_keys ? "true" : "false");
+    printf("    } qe cert extension \n");
+done:
+    free(buffer);
+    oe_cert_chain_free(&cert_chain);
+    oe_cert_free(&leaf_cert);
+}
+
+void output_certificate_chain(
+    const uint8_t* data,
+    size_t data_len,
+    bool leaf_cert_extension = true)
+{
+#if defined(__linux__)
+    const char* pem = (char*)data;
+    // in this test tool, only report buffer contains leaf cert and doesn't
```

Comment:
Did not understand this comment. Can you please clarify?

Reply:
This oecertdump outputs certificate chain in 2 scenarios:
1. Output endorsement buffer to the log file. The certificate chain in an endorsement doesn't contain a certificate that has a sgx extension. So `leaf_cert_extension` is `false`. And the buffer is output to log file so logging would be `true`. Infrastructure would need this information to be logged to help them identify the reasons for some crashes. 
2. Print report buffer to stdout in verbose mode. The leaf certificate of a report certificate chain has a sgx extension and needs to be parsed. So `leaf_cert_extension` is `false`. And the buffer is output to stdout so logging would be `false`.
Therefore, I used `bool logging = !leaf_cert_extension;` to fit the logic. It's just a special design only for this tool trying to fit the case here so we don't need extra parameters.
I will make the code comment more clear in next commit.